openc3/spec/models/widget_model_spec.rb

```diff
@@ -0,0 +1,182 @@
+# encoding: ascii-8bit
+
+# Copyright 2022 Ball Aerospace & Technologies Corp.
+# All Rights Reserved.
+#
+# This program is free software; you can modify and/or redistribute it
+# under the terms of the GNU Affero General Public License
+# as published by the Free Software Foundation; version 3 with
+# attribution addendums as found in the LICENSE.txt
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+
+# Modified by OpenC3, Inc.
+# All changes Copyright 2022,2024 OpenC3, Inc.
+# All Rights Reserved
+#
+# This file may also be used under the terms of a commercial license
+# if purchased from OpenC3, Inc.
+
+require 'spec_helper'
+require 'openc3/config/config_parser'
+require 'openc3/models/widget_model'
+
+module Aws
+  autoload(:S3, 'openc3/utilities/s3_autoload.rb')
+end
+
+module OpenC3
+  describe WidgetModel, type: :model do
+
+    before(:each) do
+      mock_redis()
+      model = ScopeModel.new(name: "DEFAULT")
+      model.create
+    end
+
+    describe "new" do
+      it "creates a widget model instance" do
+        model = WidgetModel.new(name: 'widget1', scope: 'scope')
+        expect(model).to_not be_nil
+      end
+    end
+
+    describe "all_scopes" do
+      it "exist" do
+       model = WidgetModel.new(name: 'widget1', scope: 'scope')
+       expect(WidgetModel.all_scopes).to_not be_nil
+      end
+    end
+
+    describe "self.get" do
+      it "returns the specified scope" do
+        default_time = Time.now.to_nsec_from_epoch
+        model = WidgetModel.new(name: "DEFAULT", scope: 'scope')
+        model.create
+        sleep 0.1
+        other_time = Time.now.to_nsec_from_epoch
+        model = WidgetModel.new(name: "OTHER", scope: 'scope')
+        model.create
+        sleep 0.1
+        target = WidgetModel.get(name: "DEFAULT", scope: 'scope')
+        expect(target["name"]).to eql "DEFAULT"
+        expect(target["updated_at"]).to be_within(10_000_000).of(default_time)
+        target = WidgetModel.get(name: "OTHER", scope: 'scope')
+        expect(target["name"]).to eql "OTHER"
+        expect(target["updated_at"]).to be_within(10_000_000).of(other_time)
+      end
+    end
+
+    describe "handle_config" do
```

Comment:
interface_mode_spec.rb has examples of handling both the class and instance methods for handle_config

Reply:
Found and used them, thanks.